contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,224 @@
+    // ------------
+
+    /// @notice Address of EasyTrack
+    EasyTrack public immutable easyTrack;
+
+    /// @notice Length of period in months
+    uint256 internal periodDurationMonth;
+
+    /// @notice End of the current period
+    uint256 internal currentPeriodEnd;
+
+    /// @notice The maximum that can be spent in a period
+    uint256 internal limit;
+
+    /// @notice Amount already spent in the period
+    uint256 internal spent;
+
+    // ------------
+    // CONSTRUCTOR
+    // ------------
+    constructor(EasyTrack _easy_track, address[] memory _setLimitParametersRoleHolders) {
+        easyTrack = _easy_track;
+        for (uint256 i = 0; i < _setLimitParametersRoleHolders.length; i++) {
+            _setupRole(SET_LIMIT_PARAMETERS_ROLE, _setLimitParametersRoleHolders[i]);
+        }
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Checks if _payoutSum is less than may be spent in the period
+    /// @param _payoutSum Motion sum
+    /// @return True if _payoutSum is less than may be spent in the period
+    function isUnderSpendableBalance(uint256 _payoutSum) external view returns (bool) {
+        uint256 _motionDuration = easyTrack.motionDuration();
+        if (block.timestamp + _motionDuration >= currentPeriodEnd) {
+            return _payoutSum <= limit;
+        } else {
+            return _payoutSum <= limit - spent;
+        }
+    }
+
+    /// @notice Checks if _payoutSum is less than may be spent,
+    /// @notice updates period if needed and increases the amount spent in the current period
+    function updateSpendableBalance(uint256 _payoutSum) external {
+        require(msg.sender == address(easyTrack.evmScriptExecutor()), ERROR_CALLER_IS_FORBIDDEN);
+
+        _checkAndUpdateLimitParameters();
+        _checkLimit(_payoutSum);
+        _increaseSpent(_payoutSum);
+
+        (
+            uint256 _alreadySpentAmount,
+            uint256 _spendableAmount,
+            uint256 _periodStartTimestamp,
+            uint256 _periodEndTimestamp
+        ) = this.getCurrentPeriodState();
+
+        emit FundsSpent(
+            _alreadySpentAmount,
+            _spendableAmount,
+            _periodStartTimestamp,
+            _periodEndTimestamp
+        );
+    }
+
+    /// @notice Returns balance that can be spent in the current period
+    /// @return Balance that can be spent in the current period
+    function currentSpendableBalance() external view returns (uint256) {
+        return limit - spent;
+    }
+
+    /// @notice Sets periodDurationMonth and limit
+    /// currentPeriodEnd will be calculated as a calendar date of the beginning of next month,
+    /// bi-months, quarter, half year, or year period.
+    /// @param _limit Limit to set
+    /// @param _periodDurationMonth  Length of period in months to set.
+    /// Duration of the period can be 1, 2, 3, 6 or 12 months.
+    function setLimitParameters(uint256 _limit, uint256 _periodDurationMonth)
```

Comment:
don't think we'll need to change the `spent`.
To avoid math underflow error, I added a check to this method:
```
if (spent > limit) {
    spent = limit;
}
```